src/zookeeperLock.ts

```diff
@@ -759,24 +761,27 @@ export class ZookeeperLock extends EventEmitter {
 
                             const sequence = this.filterLocks(locks)
                                 .map((l) => {
-                                    return ZookeeperLock.getSequenceNumber(l);
+                                    return {
+                                        key: l,
+                                        sequence: ZookeeperLock.getSequenceNumber(l)
+                                    };
                                 })
                                 .filter((l) => {
-                                    return l >= 0;
+                                    return l.sequence >= 0;
                                 });
 
                             debuglog(`${path}/${this.key}: lock sequence: ${JSON.stringify(sequence)}`);
 
                             const mySeq = ZookeeperLock.getSequenceNumber(this.key);
 
-                            const sorted : number[] = sequence.sort((a, b) => {
-                                return a - b;
+                            const sorted : Array<{sequence : number, key : string}> = sequence.sort((a, b) => {
+                                return a.sequence - b.sequence;
                             });
                             const offset = Math.min(sorted.length, this.config.maxConcurrentHolders);
                             const min = sorted[offset - 1];
 
                             debuglog(`${path}/${this.key}: checking ${mySeq} less than ${min} + ${this.config.maxConcurrentHolders}`);
-                            if (mySeq <= min) {
+                            if (mySeq <= min.sequence) {
```

Comment:
Since i support a mode where there can be multiple lock holders, I think it has to be this check to account for that